pkg/operator/options/suite_test.go

```diff
@@ -0,0 +1,274 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options_test
+
+import (
+	"context"
+	"flag"
+	"os"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	. "knative.dev/pkg/logging/testing"
+
+	"github.com/Azure/karpenter/pkg/apis/settings"
+	"github.com/Azure/karpenter/pkg/operator/options"
+	"github.com/Azure/karpenter/pkg/test"
+	coreoptions "github.com/aws/karpenter-core/pkg/operator/options"
+)
+
+var ctx context.Context
+
+func TestAPIs(t *testing.T) {
+	ctx = TestContextWithLogger(t)
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Options")
+}
+
+var _ = Describe("Options", func() {
+	var envState map[string]string
+	var environmentVariables = []string{
```

Comment:
this seems useful for other tests and other values that are set via env variables like around cilium, wonder if we should test these envs are set. 

Also would be good to validate that we do properly prepare the environment in this test: https://github.com/Azure/karpenter/blob/main/pkg/providers/instancetype/suite_test.go#L117

Reply:
Since those envs are from `pkg/auth/config`, maybe we could improve the test in that area.

Reply:
Since those envs are from `pkg/auth/config`, maybe we could improve the test in that area later.